src/node_buffer.cc

```diff
@@ -105,31 +103,29 @@ void CallbackInfo::Free(char* data, void*) {
 
 
 CallbackInfo* CallbackInfo::New(Isolate* isolate,
-                                Local<Object> object,
+                                Local<ArrayBuffer> object,
                                 FreeCallback callback,
                                 void* hint) {
   return new CallbackInfo(isolate, object, callback, hint);
 }
 
 
-void CallbackInfo::Dispose(Isolate* isolate) {
-  WeakCallback(isolate, PersistentToLocal(isolate, persistent_));
-}
-
-
-Persistent<Object>* CallbackInfo::persistent() {
-  return &persistent_;
-}
-
-
 CallbackInfo::CallbackInfo(Isolate* isolate,
-                           Local<Object> object,
+                           Local<ArrayBuffer> object,
                            FreeCallback callback,
                            void* hint)
     : persistent_(isolate, object),
       callback_(callback),
       hint_(hint) {
-  persistent_.SetWeak(this, WeakCallback);
+  ArrayBuffer::Contents obj_c = object->GetContents();
+  char* const data = static_cast<char*>(obj_c.Data());
+  if (object->ByteLength() != 0)
+    CHECK_NE(data, nullptr);
+
+  object->SetAlignedPointerInInternalField(0, data);
```

Comment:
Can we maybe make a comment in the header that the first internal field is reserved by node? Since switching to using array buffer I've been writing experimental modules that also make use of these fields.